confluent_kafka/schema_registry/protobuf.py

```diff
@@ -0,0 +1,437 @@
+    |                       |          | Defaults to True.                                    |
+    +-----------------------|----------+------------------------------------------------------+
+    |                       |          | Callable(SerializationContext, str) -> str           |
+    |                       |          |                                                      |
+    | subject.name.strategy | callable | Instructs the ProtobufSerializer on how to construct |
+    |                       |          | Schema Registry subject names.                       |
+    |                       |          | Defaults to topic_subject_name_strategy.             |
+    +-----------------------+----------+------------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
+    +--------------------------------------+------------------------------+
+    | Subject Name Strategy                | Output Format                |
+    +======================================+==============================+
+    | topic_subject_name_strategy(default) | {topic name}-{message field} |
+    +--------------------------------------+------------------------------+
+    | topic_record_subject_name_strategy   | {topic name}-{record name}   |
+    +--------------------------------------+------------------------------+
+    | record_subject_name_strategy         | {record name}                |
+    +--------------------------------------+------------------------------+
+
+    See ``Subject name strategy`` for additional details.
+
+    Args:
+        schema_registry_client (SchemaRegistryClient): Schema Registry
+            client instance.
+
+        descriptor (Descriptor): Protobuf Message Descriptor.
+
+        conf (dict): ProtobufSerializer configuration.
+
+    .. _Subject name strategy:
+        https://docs.confluent.io/current/schema-registry/serializer-formatter.html#subject-name-strategy
+
+    .. _Protobuf API reference:
+        https://googleapis.dev/python/protobuf/latest/google/protobuf.html
+
+    """  # noqa: E501
+    __slots__ = ['_auto_register', '_registry', '_known_subjects',
+                 'msg_index', '_proto_file', '_schema', '_schema_id',
+                 '_ref_reference_subject_func', '_subject_name_func']
+    # default configuration
+    _default_conf = {
+        'auto.register.schemas': True,
+        'subject.name.strategy': topic_subject_name_strategy,
+        'reference.subject.name.strategy': reference_subject_name_strategy
+    }
+
+    def __init__(self, schema_registry_client, descriptor, conf=None):
```

Comment:
i guess there's an adequate path to making the api support multiple types in the future, and that's to give `descriptor` a default value of `None` here. however it forces you to name the `conf` parameter.

i think we should just do the more general thing now as i think it's not too much more effort.

Reply:
see previous comment